### ISSUE TITLE
NoFollow problems in paginations.

### ISSUE THREAD

**Issue by jlmora** [PrestaShop/PrestaShop#17516]
**Describe the bug**
Bug:
1. You are pointing to paginated links with rel = “nofollow”.
2. In paginations canonicals always point to first (zero) page.
3. In paginations next/prev tag has and error. Next is pointing to self.

With this set up a bot will have problems to read paginated content and following products.

I recommend not to use nofollow in links, point canonical to self and not to use next/prev. 

**To Reproduce**
1. Go to any paginated cat
2. Click on inspect element at paginated inlink.
3. See error.
4. Go to any pagination page.
5. Click on inspect element to see html header.
6. Search for canonical and next.
4. See errors.

**Additional information**
PrestaShop version: 1.7.6.1
PHP version: N/A

**Comment by khouloudbelguith** [comment]
Hi @jlmora,

Thanks for your report.
Isn't similar to this issue: https://github.com/PrestaShop/PrestaShop/issues/13737?

Related to: https://github.com/PrestaShop/PrestaShop/issues/9503

Thanks!

**Comment by jlmora** [comment]
No. The principal problem in my report is to link paginated elements with nofollow. Maybe #13737 could help with second problem in canonical but i need to study it best.

**Comment by khouloudbelguith** [comment]
@shakaran,
I reproduced only the "nofollow link to pagination"

>Go to any paginated cat
Click on inspect element at paginated inlink.
See error.

I have the same behavior with Ps1.7.6.3 & PS1.7.5.2
![image](https://user-images.githubusercontent.com/16067358/73932357-7454b400-48da-11ea-8ff8-c91ff5f5043e.png)

Ping @PrestaShop/prestashop-product-team @jolelievre what do you think?

>Go to any pagination page.
Click on inspect element to see html header.
Search for canonical and next.
See errors.

The `rel=canonical` exists in the html header (reproduced with PS1.7.6.3 & PS1.7.5.2)
![image](https://user-images.githubusercontent.com/16067358/73933035-e2e64180-48db-11ea-8d35-a9eeb8a8d7fa.png)

But the `rel="next"` & `rel="prev" `do not exit in the html header
![image](https://user-images.githubusercontent.com/16067358/73933200-38225300-48dc-11ea-8575-a5b78b26eeab.png)

Thanks!

**Comment by jlmora** [comment]
Thanks, the only problem I see in your case is the nofollow link to pagination. My versión is 1.7.6.1 and i see the same problem in another 3 Prestashop 1.7.

**Comment by khouloudbelguith** [comment]
@jlmora, 
>Thanks, the only problem I see in your case is the nofollow link to pagination.

I'm using the theme by default Classic.

Thanks!

**Comment by jlmora** [comment]
We are using Warehouse theme.

**Comment by khouloudbelguith** [comment]
@jlmora, so, it could be a theme issue.
Thanks to check with the theme author & feedback.

**Comment by jlmora** [comment]
No, in other case i saw just now we are using Classic.

How can i send you a private text with domains? i can not make it publics.

**Comment by khouloudbelguith** [comment]
@jlmora,

My address mail: khouloud.belguith@prestashop.com
Let's be safe!

Thanks!

**Comment by eternoendless** [comment]
ping @PrestaShop/prestashop-product-team

**Comment by khouloudbelguith** [comment]
Hi @PrestaShop/prestashop-product-team @jolelievre,

As discussed with @jlmora, the issue reported & reporuduced is:
>You are pointing to paginated links with rel = “nofollow”.



Other issues are related to the custom theme

> In paginations canonicals always point to first (zero) page.
> In paginations next/prev tag has and error. Next is pointing to self.

Thanks!

**Comment by jlmora** [comment]
Thanks!

Can you ping here theme author or I need to contact directly?

**Comment by khouloudbelguith** [comment]
@jlmora, As it's not the default theme, provided by Prestashop, we can't do anything with this. 
You should contact the theme author via your addons.prestashop.com account, they are the only ones to help you with it.
Thanks!

**Comment by jlmora** [comment]
Thanks!

I will contact directly. About nofollow I await for your feedback.

**Comment by PierreRambaud** [comment]
There is a nofollow on numbers but prev & next are correctly settled.
We are waiting for SEO experts and Product team to decide what we have to do :) 
@PrestaShop/product-team

**Comment by jlmora** [comment]
Thanks for your answer. 

I hope your SEO team appreciates the importance of being able to choose whether the links to paged are follow or no follow in some strategies.

About next and prev are not taken in consideration by Google same time ago.

Saludos.

#———
# Juan Luís Mora Blanco.
# Conóceme aquí -> https://juanluismora.es/
#___


El 23 nov 2020, a las 17:50, GoT <notifications@github.com> escribió:

﻿
There is a nofollow on numbers but prev & next are correctly settled.
We are waiting for SEO experts and Product team to decide what we have to do :)
@PrestaShop/product-team

**Comment by boo-code** [comment]
The confirmed part of this report no longer applies to the default theme, and what is left is in another repository.

Measured on a 9.2 shop running hummingbird, which is the default front office theme since 9.0. Requesting a paginated category listing:

```
pagination block          0 rel= attributes, no nofollow
document head             <link rel="next"> present
```

`themes/hummingbird/templates/_partials/pagination.tpl` puts no `rel` on the numbered page links at all, and a separate `pagination-seo.tpl` emits the `rel="prev"` / `rel="next"` head links. So neither half of what was reproduced here in 2020 is present on a default 9.x install.

The classic theme still does it. `PrestaShop/classic-theme`, branch `develop`, `templates/_partials/pagination.tpl` line 44:

```smarty
rel="{if $page.type === 'previous'}prev{elseif $page.type === 'next'}next{else}nofollow{/if}"
```

That matches what @PierreRambaud described in 2020, numbers carrying `nofollow` while prev and next are set correctly. Since `themes/classic` is now its own repository and is not part of the core tree, a change there would be a classic-theme pull request rather than a core one.

One clarification in case it saves someone a false positive: a hummingbird category page does contain `nofollow`, 43 times on the page I measured, but all of those are faceted search filter links (`?q=Categories-…`, `?q=Dimension-…`) and account links, not pagination. Searching the page for the word alone will look like the bug is still there.

That leaves the decision @PierreRambaud was waiting on, whether numbered pagination links should be followable, which is unchanged and still applies to classic.
